packages/components/src/Dialog/Backdrop.tsx

```diff
@@ -38,32 +38,24 @@ export interface BackdropProps
 
 // Backdrop styles are applied here (rather than using the inline `style={...}` prop) to ensure that
 // transitions will still apply to backdrop
-export const Backdrop = styled.div.attrs((props: BackdropProps) => ({
-  backgroundColor: props.visible ? props.backgroundColor : 'transparent',
+export const Backdrop = styled.div.attrs(() => ({
```

Comment:
I think you can just pass the object instead of a callback:
```suggestion
export const Backdrop = styled.div.attrs({
```

Reply:
In v5 of Styled Components the sybfunction form is removed (it's deprecated but supported in v4):

https://styled-components.com/docs/faqs 

> Note: The subfunction object-form .attrs({ prop: props => {} }) syntax that was deprecated in v4 is removed in v5. Use function-form attrs instead .attrs(props => ({})) (you should have been seeing console warnings to make this update ahead of time.)